go/vt/schemadiff/table_test.go

```diff
@@ -1865,3 +1965,106 @@ func TestNormalize(t *testing.T) {
 		})
 	}
 }
+
+func TestIndexesCoveringForeignKeyColumns(t *testing.T) {
+	sql := `
+		create table t (
+			id int,
+			a int,
+			b int,
+			c int,
+			d int,
+			e int,
+			z int,
+			primary key (id),
+			key ax (a),
+			key abx (a, b),
+			key bx (b),
+			key bax (b, a),
+			key abcdx (a, b, c, d),
+			key dex (d, e)
+		)
+	`
+	tt := []struct {
+		columns []string
+		indexes []string
+	}{
+		{},
+		{
+			columns: []string{"a"},
+			indexes: []string{"ax", "abx", "abcdx"},
+		},
+		{
+			columns: []string{"b"},
+			indexes: []string{"bx", "bax"},
+		},
+		{
+			columns: []string{"c"},
+		},
+		{
+			columns: []string{"d"},
+			indexes: []string{"dex"},
+		},
+		{
+			columns: []string{"e"},
+		},
+		{
+			columns: []string{"z"},
+		},
+		{
+			columns: []string{"a", "b"},
+			indexes: []string{"abx", "abcdx"},
+		},
+		{
+			columns: []string{"a", "b", "c"},
+			indexes: []string{"abcdx"},
+		},
+		{
+			columns: []string{"a", "b", "c", "d"},
+			indexes: []string{"abcdx"},
+		},
+		{
+			columns: []string{"a", "b", "c", "d", "e"},
+		},
+		{
+			columns: []string{"b", "a"},
+			indexes: []string{"bax"},
+		},
+		{
+			columns: []string{"d", "e"},
+			indexes: []string{"dex"},
+		},
+		{
+			columns: []string{"a", "e"},
+		},
+	}
+
+	stmt, err := sqlparser.ParseStrictDDL(sql)
+	require.NoError(t, err)
+	createTable, ok := stmt.(*sqlparser.CreateTable)
+	require.True(t, ok)
+	c, err := NewCreateTableEntity(createTable)
+	require.NoError(t, err)
+	tableColumns := map[string]sqlparser.IdentifierCI{}
+	for _, col := range c.CreateTable.TableSpec.Columns {
+		tableColumns[col.Name.String()] = col.Name
```

Comment:
Should this be using `strings.ToLower()` as well? And should we extend the tests to have some more case mismatch cases to verify the behavior when those exist?

Reply:
Good catch, fixed, and added a test!